src/librustdoc/html/static/js/main.js

```diff
@@ -824,17 +838,59 @@ function preLoadCss(cssUrl) {
         wrapper.style.visibility = "";
         window.CURRENT_TOOLTIP_ELEMENT = wrapper;
         window.CURRENT_TOOLTIP_ELEMENT.TOOLTIP_BASE = e;
+        clearTooltipHoverTimeout(window.CURRENT_TOOLTIP_ELEMENT);
+        wrapper.onpointerenter = function(ev) {
+            // If this is a synthetic touch event, ignore it. A click event will be along shortly.
+            if (ev.pointerType !== "mouse") {
+                return;
+            }
+            clearTooltipHoverTimeout(e);
+        };
         wrapper.onpointerleave = function(ev) {
             // If this is a synthetic touch event, ignore it. A click event will be along shortly.
             if (ev.pointerType !== "mouse") {
                 return;
             }
-            if (!e.TOOLTIP_FORCE_VISIBLE && !elemIsInParent(event.relatedTarget, e)) {
-                hideTooltip(true);
+            if (!e.TOOLTIP_FORCE_VISIBLE && !elemIsInParent(ev.relatedTarget, e)) {
+                // See "Tooltip pointer leave gesture" below.
+                setTooltipHoverTimeout(e, false);
+                addClass(wrapper, "fade-out");
             }
         };
     }
 
+    function setTooltipHoverTimeout(element, show) {
```

Comment:
Could you add a JSDoc comment to this and clearTooltipHoverTimeout? In particular I'm interested seeing that `show` is documented to indicate that `true` means "show the tooltip" and `false` means "hide the tooltip".

Alternately, maybe it would be clearer if this took a callback, either `showTooltip` or `hideTooltip`? That would make it obvious at the callsite what's happening. That would require refactoring `hideTooltip` slightly so that it can take an element as its only parameter (which it might ignore). That in turn would probably require providing a `hideTooltipAndFocusBase` to cover the `hideTooltip(true)` behavior.

Broadly speaking the intuition I'm working from is that boolean parameters often wind up confusing because the meaning is not obvious at the call site.